src/index.tsx

```diff
@@ -14,9 +15,19 @@ export type CarouselProps = {
 	readonly nextButtonContent?: string | React.ReactNode;
 	readonly prevButtonContent?: string | React.ReactNode;
 	readonly showControls?: boolean;
+	readonly slideOnClick?: boolean;
+	readonly ref?: React.ForwardedRef<CarouselRef>;
 };
 
-export const Carousel: React.FC<CarouselProps> = (props: CarouselProps) => {
+export type CarouselRef = {
+	readonly next: () => void;
+	readonly prev: () => void;
+	getItems: () => CarouselItem[];
```

Comment:
Please make these properties `readonly`. You can also use the `Readonly<T>` if you don't want to explicitly write it for each prop.